File: client/src/app/modules/miscellaneous/services/competent-authority-type.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/appRest.service';
import { CompetentAuthorityType } from '../../miscellaneous/model/CompetentAuthorityType';

@Injectable({
  providedIn: 'root'
})
export class CompetentAuthorityTypeService extends AppRestService<CompetentAuthorityType>{

  constructor(http: HttpClient) {
    super(http, "miscellaneous");
  }

  getCompetentAuthorityTypes() {
    return this.getList(new HttpParams(), "competent-authority-types");
  }

}

File: src/main/java/com/jss/osiris/modules/miscellaneous/service/CompetentAuthorityTypeService.java
package com.jss.osiris.modules.miscellaneous.service;

import java.util.List;

import com.jss.osiris.modules.miscellaneous.model.CompetentAuthorityType;

public interface CompetentAuthorityTypeService {
    public List<CompetentAuthorityType> getCompetentAuthorityTypes();

    public CompetentAuthorityType getCompetentAuthorityType(Integer id);
}
